### ISSUE TITLE
Function reflector does not handle the case of using object-based value

### ISSUE THREAD

**Issue by ssut** [z-pattern-matching/z#44]
I have code like:

**types.js**
```js
const Types = Object.freeze({
  A: 'a',
  B: 'b',
});

module.exports = Object.freeze({
  ...Types,
  Types: Object.values(Types),
});
```

**index.js**
```js
const Types = require('./types');

const { matches } = require('z');

const match = matches('a')(
  (x = Types.A) => 'matched A',
  (x = Types.B) => 'matched B',
);

console.info(match);
```

and the expected behavior is to just work as the `x` is assigned by constant, but it seems not so just throwing an TypeError which looks like:

```
TypeError: Cannot read property 'A' of undefined
    at eval (eval at Parser.pushBuffer (C:\Users\suhun\Downloads\z-test\node_modules\js-function-reflector\argument_parser.js:96:22), <anonymous>:1:13)
    at Parser.pushBuffer (C:\Users\suhun\Downloads\z-test\node_modules\js-function-reflector\argument_parser.js:96:22)
    at Parser.parse (C:\Users\suhun\Downloads\z-test\node_modules\js-function-reflector\argument_parser.js:67:14)
    at module.exports (C:\Users\suhun\Downloads\z-test\node_modules\js-function-reflector\header_parser.js:10:21)
    at reflector (C:\Users\suhun\Downloads\z-test\node_modules\js-function-reflector\index.js:34:30)
    at module.exports (C:\Users\suhun\Downloads\z-test\node_modules\z\src\getMatchDetails.js:4:29)
    at resolveMatchFunctions (C:\Users\suhun\Downloads\z-test\node_modules\z\src\z.js:9:26)
    at functions (C:\Users\suhun\Downloads\z-test\node_modules\z\src\z.js:44:3)
    at Object.<anonymous> (C:\Users\suhun\Downloads\z-test\index.js:5:27)
    at Module._compile (internal/modules/cjs/loader.js:654:30)
```

I believe this should be filed in the *js-function-reflector* but I do it here because `z` has a dependency on it, which appears to be maintained by you.

**Comment by leonardiwagner** [comment]
Hi @ssut , it's there 2 problems:

- First, you can do make this work using `.call()` method to `z` passing the arguments in context, since any library could know where `Types` comes without doing this explicitly, so you should refactor index to:

```
const match = matches('a').call({ Types },
  (x = Types.A) => 'matched A',
  (x = Types.B) => 'matched B',
);
```

- Second problem, it's only working on `z@1.0.4` , please use this version while we don't fix this bug, thanks.

**Comment by leonardiwagner** [comment]
Hey @ssut , fixed that on `z@1.0.8`, please ensure you are using this newer version and passing `Types` on `.call` method:

```javascript
const match = matches('a').call({ Types },
  (x = Types.A) => 'matched A',
  (x = Types.B) => 'matched B',
);
```

Please report if it's everything working now, thanks!
